### PR TITLE
Feature/edit items

### PR DESCRIPTION
**Pull request by NeeleWaehnke**
I added the functionality to edit Items. I added an Edit-Button to the ItemCard component and created a form, that is displayed when clicked on the edit button

### CODE REVIEW

**Comment by andy-schwedler** [comment]
Looks great!
I like the new add button (much more prominent)

File: components/ItemCard.js [hunk 1]
```diff
@@ -1,22 +1,89 @@
...
+
+export default function ItemCard({
+  name,
+  date,
+  quantity,
+  onRemoveItem,
+  id,
+  onEditItem,
+}) {
+  const [isEdited, setIsEdited] = useState(false);
```
Comment (MerleMak) [inline components/ItemCard.js:15]:
```suggestion
  const [isEditing, setIsEditing] = useState(false);
```
As you are not done with the edit  yet, i think this should be named isEdting :)

File: components/ItemCard.js [hunk 2]
```diff
@@ -1,22 +1,89 @@
...
-      <StyledButton
-        onClick={() => {
-          onRemoveItem(id);
-        }}
-      >
-        <StyledDelete />
-      </StyledButton>
-    </StyledItem>
+    <>
+      {isEdited === true ? (
```
Comment (MerleMak) [inline components/ItemCard.js:31]:
```suggestion
      {isEditing === true ? (
```

File: components/ItemCard.js [hunk 3]
```diff
@@ -1,22 +1,89 @@
...
+  function handleEditSubmit(event) {
+    event.preventDefault();
+    const form = event.target.elements;
+    const editData = {
+      name: form.name.value,
+      date: form.date.value,
+      id: id,
+      quantity: form.quantity.value,
+    };
+    onEditItem(editData);
```
Comment (MerleMak) [inline components/ItemCard.js:25]:
```suggestion
    onEditItem(updatedItem);
```
Clearer naming :)

File: pages/index.js [hunk 4]
```diff
@@ -13,10 +13,19 @@ export default function Home() {
     setItems(items.filter((item) => item.id !== id));
   }
 
+  function handleEditItem(editItem) {
```
Comment (MerleMak) [inline pages/index.js:16]:
```suggestion
  function handleEditItem(updatedItem) {
```
As naming suggestion in ItemCard.js

File: pages/index.js [hunk 5]
```diff
@@ -13,10 +13,19 @@ export default function Home() {
...
+    const editedItem = items.filter((item) => item.id !== editItem.id);
+    setItems([editItem, ...editedItem]);
```
Comment (MerleMak) [inline pages/index.js:18]: I am not sure if this works, as the `const editedItem` actually descripes all the "Old" items which are not being edited. 
In the Setter function you should be adding the new item to the current "items" array in your state.
What you actually want to do is:
Check if there is an "old version of your `updatedItem` in your array 
- if yes, you want to replace it with the updatedItem

So what needs to go into this function?
1. create an array of your items that are updated. (e.g. `const updatedItemList`)
2. inside of this const you should map over your state-array(`items`), the items and inside of the map function body you ask
3. IF the id of your item (in your array `items`) === the id of your `updatedItem`, then you want to return the `updatedItem`(return updatedItem;).
4. if its not equal you want to return the item as it was (return item;)

After you mapped you still want to update your state in this function, where you only need to pass the updatedItemList, as you have already done all the work before :)

**Review by MerleMak — COMMENTED** [review]
Hi Neele, 
Great work, I left you some comments concerning your Edit function, check the out :)
Keep it up 🤘🏻🚀

File: pages/index.js (hunk 5 above)
Comment (NeeleWaehnke) [reply pages/index.js]: Hey Merle, the function works like that. I changed the name of the const editedItem to make it more clear which items are used

**Review by MerleMak — APPROVED** [review]

**Pull request merged**